auth/env_keychain_test.go

```diff
@@ -208,3 +200,76 @@ type providedAuth struct {
 func (p *providedAuth) Authorization() (*authn.AuthConfig, error) {
 	return p.config, nil
 }
+
+func useFallbackKeychain(t *testing.T) {
```

Comment:
To mirror what we did in the acceptance tests, maybe something like `assertUsesDefaultKeychain`?